### ISSUE TITLE
Is the input shape of nn.Linear in your attention layer correct?🤔️

### ISSUE THREAD

**Issue by yl305237731** [Zhenye-Na/DA-RNN#17]
first of all, nice work👏  I'm a freshman at pytorch , but I'm confused about two things about your code, If I understand wrong, please correct, thanks.

In pytorch docs,  the input shape of Linear should be (N,∗,H_in)[link](https://pytorch.org/docs/stable/nn.html?highlight=nn%20linear#torch.nn.Linear).  
You view the whole batch data before the nn.Linear layer, and then send it to the linear layer. In this case,  the whole batch data（your setting =128） will be treated as a batch data. 
the origin shape of x:
![image](https://user-images.githubusercontent.com/9928596/69129835-41c0c980-0aea-11ea-8ff9-882bd7c5b87a.png)

after your code
`x.view(-1, self.encoder_num_hidden * 2 + self.T - 1)` 
![image](https://user-images.githubusercontent.com/9928596/69129906-6b79f080-0aea-11ea-949e-433f4811e9c5.png)
the input data loss it's batch_size shape, although you recovered shape later. 
For your data, the Linear layer take the batch_szie data as one batch  will implicitly expand the time step, so there will be no negative impact. In my experiment, the batch data I input may be irrelevant, so the network can't effectively learn. 
Therefore, I deleted some view operations to ensure batchsize, which can basically achieve your effect and train on my dataset, thanks
